extensions/wikia/Search/WikiaSearchController.class.php

```diff
@@ -556,6 +557,22 @@ protected function handleArticleMatchTracking( Wikia\Search\Config $searchConfig
 	}
 
 	/**
+	 * Determine whether we should use "Go" search for exact title matches.
+	 *
+	 * This supports both the user preference and has backwards compatibility for
+	 * Monobook's "Go" button and "go" URL parameter on search.
+	 *
+	 * @return bool
+	 */
+	private function useGoSearch() {
+		$default = $this->getUser()->getGlobalPreference( 'enableGoSearch' ) ? '0' : 'Search';
+		// Support for Monobook's "Go" button and functionality
+		$fulltext = $this->getVal( 'fulltext', $this->getVal( 'go', $default ) );
+
+		return $fulltext === '0' || $fulltext === 'Go';
```

Comment:
No, because a user should always be able to override their preference with a manual URL parameter. This is especially important for the Monobook case where there is both a "Go" and a "Search" button.

In the new nav, it will default to the preference because we won't set a URL parameter. In the current nav right now, the fulltext parameter is set in the form based on the user preference, but manually setting the URL changes the behaviour, which is desired.